pkg/controller/gitopsconfig/controller.go

```diff
@@ -235,7 +235,7 @@ func (r *Reconciler) createJob(jobtype string, instance *gitopsv1alpha1.GitOpsCo
 		return reconcile.Result{}, xerrors.Errorf("unable to list the jobs in namespace %q: %w", instance.Namespace, err)
 	}
 	for _, j := range jobList.Items {
-		if isOwner(instance, &j) && j.Status.Active != 0 {
+		if isOwner(instance, &j) && (j.Status.Active != 0 || j.Status.StartTime.IsZero()) && instance.DeletionTimestamp.IsZero() {
```

Comment:
Based on our talk on chat, I now understand that the check of `DeletionTimestamp.IsZero` is a workaround to make TestCreatingDeleteJob pass, and I believe I understand more about the context why you did it this way. Thanks to your explanations, I also now finally understood the meaning of the idea you said before, that I didn't understand at that time — to modify Job.Status in the "fake client" used in TestCreatingDeleteJob. Based on this, I believe this idea you had was the correct solution here, so I would like to ask you to change it now, to:

 1. remove the `IsZero` check from this line
 2. in TestCreatingDeleteJob, add code that will simulate a change of the action=create Job Status to "completed", to allow Reconcile to make a new action=delete job

Reply:
Changes done